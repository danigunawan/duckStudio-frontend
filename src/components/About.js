import React from "react";
import "./../styles/About.css";
import {LinkX} from "react-scroll";

function About(){

    return(
        <div className="about">
            <h1 className="duck-studio-name">DUCK STUDIO</h1>
            <LinkX 
                to="section1" 
                activeClass="active"
                spy={true}
                smooth={true}
                offset={-100}
                duration={1000}>
                    <button className="shop-button">SHOP NOW</button>
            </LinkX>
        </div>
    );

}

export default About;
