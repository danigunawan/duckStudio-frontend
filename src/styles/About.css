.about {
  position: relative;
  display: flex; 
  flex-direction: column;
  align-items: center;
  background-color: white;
  margin-bottom: 1.5rem;
  padding: 2rem;
}

.duck-studio-name {
  font-size: 48px;
  font-weight: 300;
  text-align: center;
  width: auto;
  cursor: pointer;
}

.shop-button {
  margin-top: 1rem;
  font-size: 28px;
  font-weight: 300;
  width: 12rem;
  height: 3rem;
  border: none;
  text-align: center;
  background-color: rgb(250, 242, 250); 
  -webkit-filter: drop-shadow(5px 5px 5px #c5c5c5);
  filter: drop-shadow(5px 5px 5px #c5c5c5);
  cursor: pointer;
}

.shop-button:hover {
  transition: 0.1s;
  background-color: rgb(252, 239, 252);
  -webkit-filter: drop-shadow(5px 5px 5px rgb(250, 242, 250));
  filter: drop-shadow(5px 5px 5px rgb(250, 242, 250));
}

@media only screen and (max-width: 600px) {

  .duck-studio-name {
    font-size: 24px;
    font-weight: 300;
    text-align: center;
    width: auto;
    cursor: pointer;
  }

  .shop-button {
  margin-top: 1rem;
  font-size: 20px;
  font-weight: 300;
  width: 10rem;
  height: 2.5rem;
  border: none;
  text-align: center;
  background-color: rgb(250, 242, 250); 
  -webkit-filter: drop-shadow(5px 5px 5px #c5c5c5);
  filter: drop-shadow(5px 5px 5px #c5c5c5);
  cursor: pointer;
}

}