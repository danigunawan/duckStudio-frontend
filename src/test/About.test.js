import React from "react";
import ReactDOM from "react-dom";
import About from "../components/About";

import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

 it("renders without crashing", () => {
     const div = document.createElement("div");
     ReactDOM.render(<About />, div);
 });

 it("renders About correctly", () => {
    const {getByTestId} = render(<About className="about"/>);
    expect(getByTestId("about")).toHaveTextContent("h1");
});

it("matches snapshot", () => {
    const tree = renderer.create(<About className="about" />).toJSON();
    expect(tree).toMatchSnapshot();
});





































// import React from 'react';
// import { configure, shallow, render} from 'enzyme';
// import chai, { expect } from 'chai';
// import About from '../components/About';
// import chaiEnzyme from 'chai-enzyme';

// import Adapter from 'enzyme-adapter-react-16';

// configure({ adapter: new Adapter() });

// describe('<About>', function() {

//   it('should have a button', () => {
//     const wrapper = shallow(<About />);
//     expect(wrapper.find('button')).to.have.length(1);
// });

//   chai.use(chaiEnzyme())

// })