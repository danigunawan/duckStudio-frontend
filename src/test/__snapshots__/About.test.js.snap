// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="about"
  data-testid="about"
>
  <h1
    className="duck-studio-name"
  >
    DUCK STUDIO
  </h1>
  <a
    onClick={[Function]}
  >
    <button
      className="shop-button"
    >
      SHOP NOW
    </button>
  </a>
</div>
`;
